nmdc/messages.go

```diff
@@ -605,19 +614,27 @@ func (m *MyInfo) unmarshalTag(tag []byte) error {
 		i = bytes.Index(field, []byte(":"))
 		if i < 0 && r < 1 {
 			client = field
-		} else if i >= 0 {
+			continue
+		}
+		if i < 0 {
+			if len(field) == 0 {
```

Comment:
When `len(field) == 0`, `i` is always `< 0`, so this if should be even before the `i = bytes.Index(field, []byte(":"))` line.